pkg/monitor/controller.go

```diff
@@ -365,3 +369,38 @@ func (monitor *controllerMonitor) deleteAgentCRD(name string) error {
 	}
 	return nil
 }
+
+func (monitor *controllerMonitor) antreaAgentInfoAPIAvailable(stopCh <-chan struct{}) bool {
+	groupVersion := v1beta1.SchemeGroupVersion.String()
+	checkFunc := func() (done bool, err error) {
+		resources, err := monitor.client.Discovery().ServerResourcesForGroupVersion(groupVersion)
+		if err != nil {
+			if !errors.IsNotFound(err) {
+				return false, err
+			}
+			klog.InfoS("No server resources found for GroupVersion", "groupVersion", groupVersion)
+			return false, nil
+		}
+		for _, resource := range resources.APIResources {
+			if resource.Kind == agentInfoResourceKind {
+				return true, nil
+			}
+		}
+		return false, nil
+	}
+
+	found := false
+	if err := wait.PollImmediateUntil(time.Minute, func() (done bool, err error) {
```

Comment:
1 minute may be too long to wait for retry, which may make the program recover too slowly on transient errors. A shorter time like 10s should be fine.

Reply:
updated.